### PR TITLE
Add printflike to all printf like functions

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
Follow the similar kernel change: https://github.com/apache/incubator-nuttx/pull/6632
 
## Impact
Enable more compiler time check

## Testing
Pass CI

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
Please ignore the mix case warning in:
https://github.com/apache/incubator-nuttx-apps/runs/7376553170?check_suite_focus=true

**Review by pkarashchenko — APPROVED** [review]

**Comment by pkarashchenko** [comment]
There still are few build errors

**Comment by xiaoxiang781216** [comment]
Fixed.

**Comment by pkarashchenko** [comment]
```
In file included from nsh_syscmds.c:38:
nsh_syscmds.c: In function 'cmd_pmconfig':
Error: nsh_syscmds.c:220:24: error: format '%d' expects argument of type 'int', but argument 3 has type 'char *' [-Werror=format=]
  220 |       nsh_output(vtbl, "Last state %d, Next state %d",
      |                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  221 |                  "PM stay [%d, %d, %d, %d]\n",
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                  |
      |                  char *
nsh_console.h:53:49: note: in definition of macro 'nsh_output'
   53 | # define nsh_output(v, ...)    (v)->output(v, ##__VA_ARGS__)
      |                                                 ^~~~~~~~~~~
```

**Review by pkarashchenko — APPROVED** [review]
LGTM!

File: nshlib/nsh_netcmds.c [hunk 1]
```diff
@@ -377,8 +377,8 @@ static int nsh_foreach_netdev(nsh_netdev_callback_t callback,
   if (dir == NULL)
     {
       nsh_error(vtbl,
-                "nsh: %s: Could not open %s/net (is procfs mounted?): %d\n",
-                cmd, CONFIG_NSH_PROC_MOUNTPOINT, NSH_ERRNO);
```
Comment (pkarashchenko) [inline nshlib/nsh_netcmds.c:381]: Why printing of NSH_ERRNO is removed?

Comment (xiaoxiang781216) [reply nshlib/nsh_netcmds.c:381]: NSH_ERRNO may be a string or an integer:
https://github.com/apache/incubator-nuttx-apps/blob/master/nshlib/nsh.h#L434-L444
it's hard to keep it.

Comment (xiaoxiang781216) [reply nshlib/nsh_netcmds.c:381]: Ok, I remove else keyword, so the errno will output too.

File: interpreters/bas/bas_statement.c [hunk 2]
```diff
@@ -1241,7 +1241,7 @@ struct Value *stmt_EDIT(struct Value *value)
 
   return (struct Value *)0;
 #else
-  return Value_new_ERROR(value, NOTAVAILABLE, strerror(ENOSYS));
+  return Value_new_ERROR(value, NOTAVAILABLE);
```
Comment (pkarashchenko) [inline interpreters/bas/bas_statement.c:1244]: Maybe modify `#define NOTAVAILABLE       RUNTIME+22,_("Feature not available")` to `#define NOTAVAILABLE       RUNTIME+22,_("Feature not available %s")` similar to `STRAYSUBEXIT`?
However only `strerror(ENOSYS)` is printed, so does not make sense now.

**Review by pkarashchenko — APPROVED** [review]
LGTM!

File: interpreters/bas/bas_statement.c (hunk 2 above)
Comment (xiaoxiang781216) [reply interpreters/bas/bas_statement.c:1244]: Yes, I think so.

**Comment by pkarashchenko** [comment]
@xiaoxiang781216 do you have any guess about reason of check error at examples/sotest/lib/sotest/sotest.c#L58 ?

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216 do you have any guess about reason of check error at examples/sotest/lib/sotest/sotest.c#L58 ?

that's because nxstyle consider `,` in the string as the multiple variables defined in the same line:
```
static const char g_msg3[] = "Yes, don't be a stranger!";
```

**Pull request merged**